crates/matrix-sdk/src/widget/settings/element_call.rs

```diff
@@ -95,35 +54,70 @@ impl WidgetSettings {
+    /// * `element_call_url` - the url to the app e.g. <https://call.element.io>,
+    ///   <https://call.element.dev>
+    /// * `id` - the widget id.
+    /// * `parentUrl` - The url that is used as the target for the PostMessages
+    ///   sent by the widget (to the client). For a web app client this is the
+    ///   client url. In case of using other platforms the client most likely is
+    ///   setup up to listen to postmessages in the same webview the widget is
+    ///   hosted. In this case the parent_url is set to the url of the webview
+    ///   with the widget. Be aware, that this means, the widget will receive
+    ///   its own postmessage messages. The matrix-widget-api (js) ignores those
+    ///   so this works but it might break custom implementations. So always
+    ///   keep this in mind. Defaults to `element_call_url` for the non IFrame
+    ///   (dedicated webview) usecase.
+    /// * `hide_header` - defines if the branding header of Element call should
+    ///   be hidden. (default: `true`)
+    /// * `preload` - if set, the lobby will be skipped and the widget will join
+    ///   the call on the `io.element.join` action. (default: `false`)
+    /// * `font_scale` - The font scale which will be used inside element call.
+    ///   (default: `1`)
+    /// * `app_prompt` - whether element call should prompt the user to open in
+    ///   the browser or the app (default: `false`).
+    /// * `skip_lobby` Don't show the lobby and join the call immediately.
+    ///   (default: `false`)
+    /// * `confine_to_room` Make it not possible to get to the calls list in the
+    ///   webview. (default: `true`)
+    /// * `fonts` A list of fonts to adapt to ios/android system fonts.
+    ///   (default: `[]`)
+    /// * `analytics_id` - Can be used to pass a PostHog id to element call.
+    #[allow(clippy::too_many_arguments)]
     pub fn new_virtual_element_call_widget(
-        props: VirtualElementCallWidgetOptions,
+        element_call_url: String,
+        widget_id: String,
+        parent_url: Option<String>,
+        hide_header: Option<bool>,
+        preload: Option<bool>,
+        font_scale: Option<f64>,
+        app_prompt: Option<bool>,
+        skip_lobby: Option<bool>,
+        confine_to_room: Option<bool>,
+        fonts: Option<Vec<String>>,
+        analytics_id: Option<String>,
     ) -> Result<Self, url::ParseError> {
-        let mut raw_url: Url = Url::parse(&format!("{}/room", props.element_call_url))?;
+        let mut raw_url: Url = Url::parse(&format!("{element_call_url}/room"))?;
 
         let query_params = ElementCallParams {
             user_id: url_params::USER_ID.to_owned(),
             room_id: url_params::ROOM_ID.to_owned(),
             widget_id: url_params::WIDGET_ID.to_owned(),
-            avatar_url: url_params::AVATAR_URL.to_owned(),
-            displayname: url_params::DISPLAY_NAME.to_owned(),
+            display_name: url_params::DISPLAY_NAME.to_owned(),
```

Comment:
@jplatte seeing it in the diff it indeed is sth that makes more sense on the poc branch. no avatar_url in the ElementCallParams